src/System.Private.CoreLib/shared/System/Runtime/Intrinsics/X86/Avx.PlatformNotSupported.cs

```diff
@@ -761,22 +761,22 @@ public abstract class Avx : Sse42
         /// __m128 _mm_permutevar_ps (__m128 a, __m128i b)
         ///   VPERMILPS xmm, xmm, xmm/m128
         /// </summary>
-        public static Vector128<float> PermuteVar(Vector128<float> left, Vector128<float> mask) { throw new PlatformNotSupportedException(); }
+        public static Vector128<float> PermuteVar(Vector128<float> left, Vector128<int> mask) { throw new PlatformNotSupportedException(); }
```

Comment:
Why? I thought that, elsewhere, we had explicitly kept these the same type (and we have had several PRs moving these off of the integer types and onto the same type as the first parameter)....

This definitely needs to be discussed during the next design-review session. CC. @eerhardt, @terrajobst

Reply:
We only keep the same base type for mask vectors, but here is actually not mask. That is why I suggest to rename to “control”.

Reply:
You can see some intrinsic that takes “index” vector always with integer base type.

Reply:
Ah, I see, 👍 then.